rstudio/rstudio
# Unpin `testthat` version and add `opensuse15` renv.lock

### Intent

- `testthat` was pinned in #13792 to resolve `opensuse15` build issues, however, the pinned version is causing local development issues (`./install-dependencies-osx` was failing while installing `testthat`)
- removing the version pin solved the local development issue, but `opensuse15` cannot use the latest `testthat` version `3.2.0` because R `3.5.0` is no longer supported (which `opensuse15` uses)
- this PR removes the pinned `testthat` from the common depedencies and moves the version pinning to a lockfile specific to `opensuse15`

### Approach

- use `./docker-compile.sh opensuse15 server` to create an opensuse container
- install the packages listed in `dependencies/common/install-packages`, pinning `testthat` to `3.1.10` (latest version that still supports R 3.5.0)
- generate the lockfile using `renv::snapshot(repos = "https://packagemanager.posit.co/cran/2023-10-19")`
- create the same folder structure as the bionic lockfile https://github.com/rstudio/rstudio/tree/main/dependencies/common/lockfiles/bionic
- confirm that `./install-dependencies-osx` now works locally
- confirm that `./docker-compile.sh opensuse15 server` now uses the lockfile to build the docker image

### Automated Tests

none

### QA Notes

We shouldn't see docker build failures due to `testthat`, both in our jenkins pipeline and in local dev environments.

### Documentation

none

### Checklist

~- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md`~ 
~- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)~
